app/gcp_clients.py

```diff
@@ -174,39 +174,45 @@ def __init__(self, discovery_client: discovery.Resource,
                  profile: RotationProfile):
         self.discovery_client = discovery_client
         self.profile = profile
-        self.project = profile.issuingCa.project
-        self.location = profile.issuingCa.location
-        self.ca_id = profile.issuingCa.name
+        self.project = profile.issuingPool.project
+        self.location = profile.issuingPool.location
+        self.caPool = profile.issuingPool.name
+    
 
     def issueNewCert(self, cert_id: str, public_key: bytes) -> List[str]:
         cert_body = {
             'lifetime': serializeDurationForJson(self.profile.lifetime),
             'config': {
                 'publicKey': {
                     'key': serializePemBytesForJson(public_key),
-                    'type': 'PEM_RSA_KEY',
+                    'format': 'PEM',
                 },
                 'subjectConfig': {
-                    'commonName': self.profile.dnsName,
+                    'subject': {
+                        'commonName': self.profile.dnsName,
+                    },
                     'subjectAltName': {
                         'dnsNames': [self.profile.dnsName],
                     }
                 },
-                'reusableConfig': {
-                    'reusableConfig':
-                    'projects/privateca-data/locations/{}/reusableConfigs/leaf-server-tls'
-                    .format(self.location),
+                'x509Config': {
+                    'caOptions': {
+                        'isCa': 'false'
+                        },
+                    'keyUsage': {
```

Comment:
could you explicitly include the key usages and EKUs from here: https://cloud.devsite.corp.google.com/certificate-authority-service/docs/certificate-profile#end-server-tls

specifically:
```
keyUsage:
  baseKeyUsage:
    digitalSignature: true
    keyEncipherment: true
  extendedKeyUsage:
    serverAuth: true
```